bubblegum/program/src/state/mod.rs

```diff
@@ -50,17 +50,42 @@ impl Voucher {
     }
 }
 
-#[event]
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
 pub struct NewNFTEvent {
+    pub account_type: AccountType,
     pub version: Version,
     pub metadata: MetadataArgs,
     pub nonce: u64,
 }
 
-#[event]
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
+pub struct NodeEvent {
+    pub account_type: AccountType,
+    pub version: Version,
+    // TODO: Figure out why we cannot generate JS API if this is `spl_account_compression::Node`.
+    pub node: [u8; 32],
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
 pub struct NFTDecompressionEvent {
+    pub account_type: AccountType,
     pub version: Version,
     pub id: Pubkey,
     pub tree_id: Pubkey,
     pub nonce: u64,
 }
+
+#[derive(BorshDeserialize, BorshSerialize, PartialEq, Eq, Debug, Clone)]
+#[repr(u8)]
+pub enum AccountType {
```

Comment:
Should probably be called `BubblegumEventType`

Reply:
Done in next version.